content/docs/setup/kubernetes/additional-setup/mesh-expansion/index.md

```diff
@@ -175,49 +196,115 @@ The following example shows accessing a service running in the Kubernetes cluste
+    $ echo ${GCE_IP}
+    {{< /text >}}
+
+1. Configure Service Discovery for VM by `ServiceEntry`. You add VM services to the mesh by configuring a
+[`ServiceEntry`](/docs/reference/config/networking/v1alpha3/service-entry/). A `ServiceEntry` lets you manually add
 additional services to Istio's model of the mesh so that other services can find and direct traffic to them. Each
 `ServiceEntry` configuration contains the IP addresses, ports, and labels (where appropriate) of all VMs exposing a
 particular service, as in the following example.
 
-{{< text bash yaml >}}
-$ kubectl -n test apply -f - <<EOF
-apiVersion: networking.istio.io/v1alpha3
-kind: ServiceEntry
-metadata:
-  name: vm1
-spec:
-   hosts:
-   - vm1.test.svc.cluster.local
-   ports:
-   - number: 80
-     name: http
-     protocol: HTTP
-   resolution: STATIC
-   endpoints:
-    - address: 10.128.0.17
-      ports:
-        http: 8080
-      labels:
-        app: vm1
-        version: 1
-EOF
+
+    {{< text bash yaml >}}
+    $ kubectl -n ${SERVICE_NAMESPACE} apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+    name: vmhttp
+    spec:
+    hosts:
+    - vmhttp.${SERVICE_NAMESPACE}.svc.cluster.local
+    ports:
+    - number: 8080
+        name: http
+        protocol: HTTP
+    resolution: STATIC
+    endpoints:
+        - address: ${GCE_IP}
+        ports:
+            http: 8080
+        labels:
+            app: vmhttp
+            version: "v1"
+    EOF
+    {{< /text >}}
+
+1. Workloads in a Kubernetes cluster need a DNS mapping to resolve the service running on the VM.
+You can integrate with your own DNS system. For illustration purpose, we use `istioctl register`
+which creates a Kubernetes `selector-less` service.
+
+    {{< text bash >}}
+    $ istioctl  register -n ${SERVICE_NAMESPACE} vmhttp ${GCE_IP} 8080
+    {{< /text >}}
+
+1. We deploy a sleep pod in the Kubernetes cluster, and wait for the pod to be ready.
+
+    {{< text bash >}}
+    $ kubectl apply -f samples/sleep/sleep.yaml
+    $ kubectl get po
+    NAME                             READY     STATUS    RESTARTS   AGE
+    productpage-v1-8fcdcb496-xgkwg   2/2       Running   0          1d
+    sleep-88ddbcfdd-rm42k            2/2       Running   0          1s
+    ...
+    {{< /text >}}
+
+1. Send a request from the sleep container to the VM HTTP service.
+
+    {{< text bash >}}
+    $ kubectl exec -it sleep-88ddbcfdd-rm42k -c sleep -- curl vmhttp.${SERVICE_NAMESPACE}.svc.cluster.local:8080
+    <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2 Final//EN"><html>
+    <title>Directory listing for /</title>
+    <body>
+    <h2>Directory listing for /</h2>
+    <hr>   
+    <ul>                                                                
+    <li><a href=".bashrc">.bashrc</a>
+    <li><a href=".ssh/">.ssh/</a>
+    </body>
+    {{< /text >}}
+
+## Cleanup
+
+Run the following commands to remove the expansion VM from the mesh.
```

Comment:
```suggestion
Run the following commands to remove the VM services from the mesh's abstract model.
```

Reply:
Given the page is titled as "mesh expansion" I think "remove the VM from the mesh" is clear enough? Any reason to mention "abstract model"? I'm not sure if we have settled down a model for VM.

Reply:
So, the abstract model is not the model of the VM but the abstract model Pilot maintains for the mesh. Remove VM from mesh is not false but it's too general. The added specificity helps readers understand the behavior.

Reply:
Done.